src/main/java/Duke.java

```diff
@@ -1,59 +1,122 @@
 import java.util.Scanner;
+import java.util.Arrays;
 import java.util.ArrayList;
 
 public class Duke {
 
     public static String indent = "     ";
 
+    // init array of Task objects
     public static ArrayList<Task> allTasks = new ArrayList<>();
 
+    // helper function to convert string array to string
+    // not just string rep like what Arrays.toString(strArr) does
+    public static String convertToString(String[] strArr) {
+//        StringBuilder builder = new StringBuilder();
+//        for(String s : strArr) {
+//            builder.append(s + " ");
+//        }
+//        return builder.toString();
+        return String.join(" ", strArr);
+    }
+
+    // print array of Task objects
     public static void printAllTasks() {
         for (int i = 0; i < allTasks.size(); i++) {
-            String symbol = allTasks.get(i).getIsDone() ? "\u2713" : "\u2717" ;
-            System.out.println(indent + (i + 1) + ". [" + symbol + "] " + allTasks.get(i).getTask());
+            System.out.println(indent + (i + 1) + ". " + allTasks.get(i));
         }
     }
 
-    public static void updateTaskAsDone(int num) {
-        if (num >= allTasks.size()) {
+    // print newly added task
+    public static void printNewlyAddedTask(Task task) {
+        System.out.println(indent + "Got it. I've added this task: "
+                + System.lineSeparator()
+                + indent + indent + task
+                + System.lineSeparator()
+                + indent + "Now you have " + allTasks.size() + " tasks in the list.");
+    }
+
+    // mark a Task as done
+    public static void markTaskAsDone(int num) {
+        if (num > allTasks.size() || num == 0) {
             System.out.println(indent + "Sorry, this task does not exist!");
         }
         else {
-            allTasks.get(num - 1).setAsDone();
+            allTasks.get(num - 1).setIsDone(true);
             System.out.println(indent + "Nice! I've marked this task as done:");
             System.out.println(indent + "[\u2713] " + allTasks.get(num - 1).getTask());
         }
     }
 
-    public static void addTask(String input) {
-        if (!input.isEmpty()) {
-            Task newTask = new Task(input);
-            allTasks.add(newTask);
-            System.out.println(indent + "added: " + input);
-        }
-        else {
-            System.out.println(indent + "No text specified.");
+    public static void addTask(String[] inputArr) {
+
+        /* TODO: implement error checking for user input
+           cases:
+           - input is completely empty
+           - inputArr[0] == _todo/deadline/event, but rest of the input is completely empty
+           - deadline/event does not have "/by" or "/at"
+        */
+
+        Task newObj;
```

Comment:
The Task name newObj is not very obvious later on, perhaps it could benefit code readability if it was changed to newTask?